app/models/manageiq/providers/azure_stack/inventory.rb

```diff
@@ -6,6 +6,21 @@ class ManageIQ::Providers::AzureStack::Inventory < ManageIQ::Providers::Inventor
   # Default manager for building collector/parser/persister classes
   # when failed to get class name from refresh target automatically
   def self.default_manager_name
-    'CloudManager'
+    'CloudManager' # TODO; version?
+  end
+
+  # Sets the appropriate class of versioned collector for
+  # CloudManager and NetworkManager targets
+  def self.collector_class_for(ems, target = nil, manager_name = nil)
+    target = ems if target.nil?
+
+    case target
```

Comment:
Instead `case` here I think we should just delegate `api_version` on network manager to the cloud manager, here https://github.com/manageiq/manageiq-providers-azure_stack/blob/master/app/models/manageiq/providers/azure_stack/network_manager.rb#L9

Reply:
awesome :+1: